flake.nix

```diff
@@ -36,27 +36,77 @@
     ]
       (system:
       let
-        pkgs = import inputs.nixpkgs { inherit system; };
+        compiler = "ghc964";
+
+        # nixpkgs enhanced with haskell.nix and crypto libs as used by iohk
+
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [
+            # This overlay contains libsodium and libblst libraries
+            inputs.iohk-nix.overlays.crypto
+            # This overlay contains pkg-config mappings via haskell.nix to use the
+            # crypto libraries above
+            inputs.iohk-nix.overlays.haskell-nix-crypto
+            # Keep haskell.nix as the last overlay!
+            #
+            # Reason: haskell.nix modules/overlays neds to be last
+            # https://github.com/input-output-hk/haskell.nix/issues/1954
+            inputs.haskellNix.overlay
+            # Custom static libs used for darwin build
+            (import ./nix/static-libs.nix)
+          ];
+        };
+
         tools = {
-          hlint = hydraProject.pkgs.haskell-nix.tool hydraProject.compiler "hlint" "3.8";
+          apply-refact = pkgs.haskell-nix.tool compiler "apply-refact" "0.14.0.0";
+          cabal = pkgs.haskell-nix.cabal-install.${compiler};
+          cabal-fmt = pkgs.haskell-nix.tool compiler "cabal-fmt" "0.1.9";
+          cabal-plan = pkgs.haskell-nix.tool compiler "cabal-plan" "latest";
+          fourmolu = pkgs.haskell-nix.tool compiler "fourmolu" "0.14.0.0";
+          haskell-language-server = pkgs.haskell-nix.tool compiler "haskell-language-server" rec {
+            src = inputs.hls;
+            cabalProject = builtins.readFile (src + "/cabal.project");
+          };
+          hlint = pkgs.haskell-nix.tool compiler "hlint" "3.8";
+          inherit (pkgs)
+            check-jsonschema
+            git
+            gnuplot
+            nixpkgs-fmt
+            nodejs
+            pkg-config
+            plantuml
+            treefmt
+            websocat
+            yarn
+            yq;
+          inherit (pkgs.haskellPackages) hspec-discover ghcid graphmod;
+          inherit (inputs.cardano-node.packages.${system}) cardano-node cardano-cli;
+          mithril-client-cli = inputs.mithril.packages.${system}.mithril-client-cli;
         };
-        hydraProject = import ./nix/hydra/project.nix {
-          inherit (inputs) haskellNix iohk-nix CHaP;
-          inherit system nixpkgs;
+
+        inputMap = { "https://intersectmbo.github.io/cardano-haskell-packages" = inputs.CHaP; };
```

Comment:
This seems to be a very haskell.nix specific thing.

Would it be a good idea to pass `inputs` into our nix modules (like we do for `packages.nix`) and use what we care for inside? Or this bad practice?

Reply:
My intuition here is that project.nix is a haskell.nix function and should be as transparent as possible with the arguments it expects. Exposing it with argument names that are more general feels like coupling, so I just went for inherit compiler-nix-name and inputMap.

Reply:
If horizon then we would not have a either compiler argument or a CHaP argument.